src/semantic/types.rs

```diff
@@ -325,6 +351,50 @@ impl TypeInfo {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+pub struct ClassMember {
+    name: String,
+    location: Location,
+    typ: Type,
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub struct ClassMembers(HashMap<String, ClassMember>);
+
+impl ClassMembers {
+    fn new() -> ClassMembers {
+        ClassMembers(HashMap::new())
+    }
+    fn add_member(&mut self, class_name: String, member: ClassMember) -> Result<()> {
+        let location = &member.location.clone();
+        let member_name = member.name.clone();
+        match self.0.insert(member.name.clone(), member) {
+            Some(previous_field) => Err(SemanticError::redefined_field(
+                location,
+                member_name,
+                class_name,
+                previous_field.location,
+            )),
+            None => Ok(()),
+        }
+    }
+    fn get_member(
+        &self,
+        location: &Location,
+        class_name: String,
+        name: &String,
+    ) -> Result<&ClassMember> {
+        match self.0.get(name) {
+            Some(v) => Ok(v),
+            None => Err(SemanticError::no_member_named(
+                location,
+                class_name,
+                name.clone(),
+            )),
+        }
+    }
+}
+
```

Comment:
Here are new structures to control the class member concept in the type checker.